Mathlib/Topology/ContinuousFunction/CFC.lean

```diff
@@ -0,0 +1,1111 @@
+    [h' : Subsingleton (CFCCoreClass R (r • a))] (f : C(R, R)) :
+    cfc₁ R a (f.comp (r • (ContinuousMap.id R).restrict (spectrum R a))) = cfc₂ R (r • a) f := by
+  have key : cfc₁ R a (r • (ContinuousMap.id R).restrict (spectrum R a)) = r • a := by simp
+  have : CFCCoreClass R (cfc₁ R a (r • (ContinuousMap.id R).restrict (spectrum R a))) := by
+    convert h
+  have : Subsingleton
+    (CFCCoreClass R (cfc₁ R a (r • (ContinuousMap.id R).restrict (spectrum R a)))) := by convert h'
+  convert cfc₁₂_comp a (r • (ContinuousMap.id R).restrict (spectrum R a)) f
+  exact key.symm
+
+lemma cfc₂_smul_comm (r : R) [CFCCoreClass R (r • a)] [Subsingleton (CFCCoreClass R (r • a))]
+    (f : C(R, R)) : cfc₂ R a (f.comp (r • (ContinuousMap.id R))) = cfc₂ R (r • a) f :=
+  cfc₁_smul_comm a r f
+
+lemma cfc₁_star_comm [h : CFCCoreClass R (star a)] [h' : Subsingleton (CFCCoreClass R (star a))]
+    (f : C(R, R)) :
+    cfc₁ R a (f.comp (star (ContinuousMap.id R).restrict (spectrum R a))) = cfc₂ R (star a) f := by
+  have key : cfc₁ R a (star (ContinuousMap.id R).restrict (spectrum R a)) = star a := by
+    simp [map_star]
+  have : CFCCoreClass R (cfc₁ R a (star ((ContinuousMap.id R).restrict (spectrum R a)))) := by
+    convert h
+  have : Subsingleton
+    (CFCCoreClass R (cfc₁ R a (star (ContinuousMap.id R).restrict (spectrum R a)))) := by
+      convert h'
+  convert cfc₁₂_comp a (star (ContinuousMap.id R).restrict (spectrum R a)) f
+  exact key.symm
+
+lemma cfc₂_star_comm [CFCCoreClass R (star a)] [Subsingleton (CFCCoreClass R (star a))]
+    (f : C(R, R)) : cfc₂ R a (f.comp (star (ContinuousMap.id R))) = cfc₂ R (star a) f :=
+  cfc₁_star_comm a f
+
+end CFC
+
+end Semiring
+
+section Ring
+
+variable {R A : Type*} [CommRing R] [StarRing R] [TopologicalSpace R]
+    [TopologicalRing R] [ContinuousStar R] [Ring A] [StarRing A] [TopologicalSpace A] [Algebra R A]
+    (a : A) [CFCClass R a] [h : CFCCoreClass R (-a)]
+
+
+lemma cfc₁_neg_comm [h' : Subsingleton (CFCCoreClass R (-a))] (f : C(R, R)) :
+    cfc₁ R a (f.comp (-((ContinuousMap.id R).restrict (spectrum R a)))) = cfc₂ R (-a) f := by
+  have key : cfc₁ R a (-(ContinuousMap.id R).restrict (spectrum R a)) = -a := by simp
+  have : CFCCoreClass R (cfc₁ R a (-(ContinuousMap.id R).restrict (spectrum R a))) := by
+    convert h
+  have : Subsingleton
+    (CFCCoreClass R (cfc₁ R a (-(ContinuousMap.id R).restrict (spectrum R a)))) := by convert h'
+  convert cfc₁₂_comp a (-(ContinuousMap.id R).restrict (spectrum R a)) f
+  exact key.symm
+
+lemma cfc₂_neg_comm [Subsingleton (CFCCoreClass R (-a))]
+    (f : C(R, R)) : cfc₂ R a (f.comp (-ContinuousMap.id R)) = cfc₂ R (-a) f :=
+  cfc₁_neg_comm a f
+
+end Ring
+
+section Field
+
+variable {F A : Type*} [Field F] [StarRing F] [TopologicalSpace F]
+    [TopologicalRing F] [ContinuousStar F] [Ring A] [StarRing A] [TopologicalSpace A] [Algebra F A]
+    (a : A) [CFCClass F a] [h : CFCClass F (1 : A)] [h' : Subsingleton (CFCCoreClass F (1 : A))]
```

Comment:
I don't really like the idea of using `F` for fields; in most of mathlib `E`, `F` are things like vector spaces, and `𝕜` is the field.